src/vs/workbench/contrib/preferences/browser/settingsEditorSettingIndicators.ts

```diff
@@ -106,28 +110,43 @@ export class SettingsTreeIndicatorsLabel {
 		this.scopeOverridesElement.style.display = 'none';
 		if (element.overriddenScopeList.length) {
 			this.scopeOverridesElement.style.display = 'inline';
-			const otherOverridesLabel = element.isConfigured ?
-				localize('alsoConfiguredIn', "Also modified in") :
-				localize('configuredIn', "Modified in");
-
-			DOM.append(this.scopeOverridesElement, $('span', undefined, `${otherOverridesLabel}: `));
-
-			for (let i = 0; i < element.overriddenScopeList.length; i++) {
-				const view = DOM.append(this.scopeOverridesElement, $('a.modified-scope', undefined, element.overriddenScopeList[i]));
-
-				if (i !== element.overriddenScopeList.length - 1) {
-					DOM.append(this.scopeOverridesElement, $('span', undefined, ', '));
-				}
-
+			if (element.overriddenScopeList.length === 1) {
+				// Just show all the text in the label.
+				const configuredInText = element.isConfigured ?
+					localize('alsoConfiguredIn', "Also modified in") :
+					localize('configuredIn', "Modified in");
+				this.scopeOverridesLabel.text = `${configuredInText}: `;
+
+				const firstScope = element.overriddenScopeList[0];
+				const view = DOM.append(this.scopeOverridesElement, $('a.modified-scope', undefined, firstScope));
 				elementDisposables.add(
 					DOM.addStandardDisposableListener(view, DOM.EventType.CLICK, (e: IMouseEvent) => {
 						onDidClickOverrideElement.fire({
 							targetKey: element.setting.key,
-							scope: element.overriddenScopeList[i]
+							scope: firstScope
 						});
 						e.preventDefault();
 						e.stopPropagation();
 					}));
+			} else {
+				// Show most of the text in a custom hover.
+				let scopeOverridesLabelText = '$(info) ';
+				scopeOverridesLabelText += element.isConfigured ?
+					localize('alsoConfiguredElsewhere', "Also modified elsewhere") :
+					localize('configuredElsewhere', "Modified elsewhere");
+				this.scopeOverridesLabel.text = scopeOverridesLabelText;
+
+				const scopeOverridesContent = $('.scope-overrides-hover-content');
+				const scopeOverridesHoverMessagePreface = element.isConfigured ?
+					localize('alsoModifiedInScopes', "The setting has also been modified in the following scopes:") :
+					localize('modifiedInScopes', "The setting has been modified in the following scopes:");
+				const prefaceParagraph = DOM.append(scopeOverridesContent, $('p', { 'style': 'padding: 4px 8px;' }, scopeOverridesHoverMessagePreface));
+				const listMessage = DOM.append(prefaceParagraph, $('ul'));
+				for (const scope of element.overriddenScopeList) {
+					const listBullet = DOM.append(listMessage, $('li'));
+					DOM.append(listBullet, $('span.modified-scope', undefined, scope));
+				}
+				setupCustomHover(this.hoverDelegate, this.scopeOverridesElement, scopeOverridesContent);
```

Comment:
I'm pretty sure hover lock won't work if you're not using IHoverService

Reply:
hoverDelegate calls IHoverService so hover lock still works https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/preferences/browser/settingsEditorSettingIndicators.ts#L51